src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle comparedRectangle = (Rectangle) rectangle;
+            return (color == comparedRectangle.color || color.equals(comparedRectangle.color)
+                    || (color == null && comparedRectangle.color == null))
```

Comment:
still NPE:
let's go step by step with such conditions(color = null, comparedRectangle.color = "Red"):
1) First check (color == comparedRectangle.color) => false;
2) Second check (color.equals(comparedRectangle.color)) => NullPointerException because color = null;
3) Third check can't be reached in this case;